lib/certgenerateutils/certgenerate.go

```diff
@@ -0,0 +1,109 @@
+package certgenerateutils
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+	"os"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+func generateSerial() (*big.Int, error) {
+	// According to
+	// https://cabforum.org/wp-content/uploads/CA-Browser-Forum-BR-1.6.4.pdf:
+	//
+	// Effective September 30, 2016, CAs SHALL generate
+	// non-sequential Certificate serial numbers greater than zero
+	// (0) containing at least 64 bits of output from a CSPRNG.
+	//
+	// Here, we set the limit to double this requirement.
+	limit := new(big.Int).Lsh(big.NewInt(1), 128)
+	ret, err := rand.Int(rand.Reader, limit)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to generate serial number")
+	}
+	return ret, nil
+}
+
+func GenerateCert(fqdn string) error {
```

Comment:
Instead of using os.Create in this function to create files in same path please use os.CreateTemp() to create in temporary folder used by golang.
Please return the certificate and the key path also from the function so that the caller knows about the created files and will be able to delete them

Reply:
done